cluster-autoscaler/cloudprovider/gce/gce_price_model_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	. "k8s.io/autoscaler/cluster-autoscaler/utils/test"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetNodePrice(t *testing.T) {
+	labels1, _ := buildGenericLabels(GceRef{
+		Name:    "kubernetes-minion-group",
+		Project: "mwielgus-proj",
+		Zone:    "us-central1-b"},
+		"n1-standard-8", "sillyname")
+
+	labels2, _ := buildGenericLabels(GceRef{
+		Name:    "kubernetes-minion-group",
+		Project: "mwielgus-proj",
+		Zone:    "us-central1-b"},
+		"n1-standard-8", "sillyname")
+	labels2[preemptibleLabel] = "true"
+
+	model := &GcePriceModel{}
+	now := time.Now()
+
+	// regular
+	node1 := BuildTestNode("sillyname1", 8000, 30*1024*1024*1024)
+	node1.Labels = labels1
+	price1, err := model.NodePrice(node1, now, now.Add(time.Hour))
+	assert.NoError(t, err)
+
+	// preemptable
+	node2 := BuildTestNode("sillyname2", 8000, 30*1024*1024*1024)
+	node2.Labels = labels2
+	price2, err := model.NodePrice(node2, now, now.Add(time.Hour))
+	assert.NoError(t, err)
+	assert.True(t, price1 > 3*price2)
```

Comment:
I would rather base this assert on preemptibleDiscount const or at the very least make 3 a const. I don't like an idea of unittests failing after a change in pricing model (especially if it was taken from config file as per your TODO)